control_unit.py

```diff
@@ -67,22 +67,33 @@ def _STR(instruction):
 
 
 def _RTI(instruction):
-    """unused"""
-    pass  # TODO throw exception here.
+    raise Exception("instruction is not implemented !")
+    pass  
 
 
 def _NOT(instruction):
-    """bitwise not"""
+    DR = (_NOT(instruction) >> 9) & 0X7                            # Destenation Register.
```

Comment:
`(_NOT(instruction) >> 9)`
that's an endless recursive call.
You probably mean:
`DR = (instruction >> 9) & 0X7`

Reply:
wrote that line first as you did in the comment but i thought i did it wrong

Reply:
no problem, but I don't understand why did you think calling `_NOT(instruction)` is the right thing to do?